Source/WebCore/dom/Document.cpp

```diff
@@ -10715,8 +10715,15 @@ PermissionsPolicy Document::permissionsPolicy() const
     // We create PermissionsPolicy on demand instead of at Document creation time,
     // because Document may not be set on Frame yet, and it would affect the computation
     // of PermissionsPolicy.
-    if (!m_permissionsPolicy)
-        m_permissionsPolicy = makeUnique<PermissionsPolicy>(ownerElement(), securityOrigin().data());
```

Comment:
Maybe instead, PermissionsPolicy could take a Document& and below logic would go in PermissionsPolicy constructor.